hw/mcu/nxp/kinetis/src/hal_spi.c

```diff
@@ -0,0 +1,624 @@
+        hal_cfg.sout_pin = (uint8_t)cfg->pin_miso;
+    }
+    hal_cfg.pcs_pin = (uint8_t)cfg->pin_ss;
+    return hal_spi_init(spi_num, &hal_cfg, spi_type);
+}
+
+int
+hal_spi_config(int spi_num, struct hal_spi_settings *settings)
+{
+    struct nxp_hal_spi *spi;
+    struct nxp_spi_master *master;
+    struct nxp_spi_slave *slave;
+    dspi_clock_polarity_t cpol;
+    dspi_clock_phase_t cpha;
+
+    spi = hal_spi_resolve(spi_num);
+    if (!spi) {
+        return EINVAL;
+    }
+
+    if (!settings) {
+        return EINVAL;
+    }
+
+    switch (settings->data_mode) {
+    case HAL_SPI_MODE0:
+        cpol = kDSPI_ClockPolarityActiveHigh;
+        cpha = kDSPI_ClockPhaseFirstEdge;
+        break;
+    case HAL_SPI_MODE1:
+        cpol = kDSPI_ClockPolarityActiveHigh;
+        cpha = kDSPI_ClockPhaseSecondEdge;
+        break;
+    case HAL_SPI_MODE2:
+        cpol = kDSPI_ClockPolarityActiveLow;
+        cpha = kDSPI_ClockPhaseFirstEdge;
+        break;
+    case HAL_SPI_MODE3:
+        cpol = kDSPI_ClockPolarityActiveLow;
+        cpha = kDSPI_ClockPhaseSecondEdge;
+        break;
+    default:
+        assert(0);
+        break;
+    }
+
+    if (spi->type == HAL_SPI_TYPE_MASTER) {
+        master = (struct nxp_spi_master*) spi;
+
+        master->config.ctarConfig.baudRate = settings->baudrate;
+        master->config.ctarConfig.pcsToSckDelayInNanoSec =
+            1000000000U / master->config.ctarConfig.baudRate;
+        master->config.ctarConfig.lastSckToPcsDelayInNanoSec =
+            1000000000U / master->config.ctarConfig.baudRate;
+        master->config.ctarConfig.betweenTransferDelayInNanoSec =
+            1000000000U / master->config.ctarConfig.baudRate;
+        master->config.ctarConfig.direction =
+            (settings->data_order == HAL_SPI_MSB_FIRST) ?
+            kDSPI_MsbFirst :
+            kDSPI_LsbFirst;
+        master->config.ctarConfig.bitsPerFrame =
+            (settings->word_size == HAL_SPI_WORD_SIZE_8BIT) ? 8 : 9;
+        master->config.ctarConfig.cpol = cpol;
+        master->config.ctarConfig.cpha = cpha;
+    } else {
+        slave = (struct nxp_spi_slave*) spi;
+        slave->config.ctarConfig.bitsPerFrame =
+            (settings->word_size == HAL_SPI_WORD_SIZE_8BIT) ? 8 : 9;
+        slave->config.ctarConfig.cpol = cpol;
+        slave->config.ctarConfig.cpha = cpha;
+    }
+    return 0;
+}
+
+int
+hal_spi_enable(int spi_num)
+{
+    struct nxp_hal_spi *spi;
+    struct nxp_spi_master *master;
+    struct nxp_spi_slave *slave;
+    spi = hal_spi_resolve(spi_num);
+    if (!spi) {
+        return EINVAL;
+    }
+    if (spi->enabled) {
+        return 0;
+    }
+
+    if (spi->type == HAL_SPI_TYPE_MASTER) {
+    master = (struct nxp_spi_master*) spi;
+    DSPI_MasterInit(spi->dev,
+                    &master->config,
+                    CLOCK_GetFreq(kCLOCK_BusClk));
+    } else {
+    slave = (struct nxp_spi_slave*) spi;
+    DSPI_SlaveInit(spi->dev, &slave->config);
+    }
```

Comment:
`if` and `else` blocks are indented incorrectly.

Reply:
Addressed.